packages/host/app/components/operator-mode/profile-settings-modal.gts

```diff
@@ -0,0 +1,136 @@
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+import { restartableTask } from 'ember-concurrency';
+
+import perform from 'ember-concurrency/helpers/perform';
+
+import {
+  BoxelButton,
+  BoxelInput,
+  FieldContainer,
+} from '@cardstack/boxel-ui/components';
+
+import ModalContainer from '@cardstack/host/components/modal-container';
+
+import { ProfileInfo } from '@cardstack/host/components/operator-mode/profile-info-popover';
+import config from '@cardstack/host/config/environment';
+import MatrixService from '@cardstack/host/services/matrix-service';
+
+interface Signature {
+  Args: {
+    toggleProfileSettings: () => void;
+  };
+  Element: HTMLElement;
+}
+
+export default class ProfileSettingsModal extends Component<Signature> {
+  @service declare matrixService: MatrixService;
+
+  @tracked private displayName: string | undefined = undefined;
+  @tracked private saveSuccessIndicatorShown = false;
+
+  @action setDisplayName(name: string) {
+    this.displayName = name;
+  }
+
+  private saveTask = restartableTask(async () => {
+    await this.matrixService.profile.loaded; // Prevent saving before profile is loaded
+    let delayMs = config.environment === 'test' ? 1 : 1000;
```

Comment:
let's just put the delay in the config and have the delay for the test env be 0. Simliar to these: 
https://github.com/cardstack/boxel/blob/main/packages/host/config/environment.js#L73-L77

it's was simpler that way.

Reply:
Ah I see, I adjusted this.